db/migrate/20170913091115_create_last_scan_result.rb

```diff
@@ -0,0 +1,9 @@
+class CreateLastScanResult < ActiveRecord::Migration[5.0]
+  def change
+    create_table :last_scan_results do |t|
+      t.string :last_scan_status
+      t.string :last_scan_message
+      t.belongs_to :resource, :polymorphic => true, :type => :bigint
```

Comment:
I would link this to the job that created the status. We could get the target resource through that. Is that enough to have the needed info of scan_type @roliveri ?
(I know that for `ContainerManager::Scanning::Job` we always have OpenSCAP AND package analysis)

Reply:
I don't think so. Assuming the scan type based on the resource won't be enough. As I mentioned, going forward, different scans may be performed independently for the same resource, so they may be performed by different jobs. Even if we could obtain the scan type through the job, do job objects persist after the jobs have completed? If not, the link to the job could go stale while the scan results are still linked to the resource.

Reply:
Can we have different scan types in a vm scanning job? Anyway regardless the new table might be a better place to keep the `scan_type` (and anything related to `scan` and not directly to `job`)

I still think the a link to the job would serve us (for example if we want to show last job info in container image screens). I don't see a purge or something that would delete job records but I might be missing something.

Reply:
We haven't had different scan types until now, so there's no concept of type in the scan job. However, I think we may need to rely on subclasses of Job more than we do now. The Job subclass could not only indicate the resource being scanned, but the type of scan as well. If the scanning process is significantly different for some types, it makes sense for them to have their own Job subclass.

I'm not saying we shouldn't link to Job. Only that we should make sure it will be there when we need it.